### PR TITLE
Vih 9985 nunit automation test

### PR DESCRIPTION
**Pull request by phildebrah**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-9985

### Change description ###

Creating a Booking hearing with Video Access (End point test)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: UI.NUnitVersion/TestData/HearingTestData.cs [hunk 1]
```diff
@@ -47,4 +47,43 @@ public static BookingDto CreateHearingDto(bool remote = false, DateTime? schedul
         };
         return bookingDto;
     }
+
+    public static BookingDto CreateHearingEndPoint(bool remote = false, DateTime? scheduledDateTime = null)
```
Comment (shaed-parkar) [inline UI.NUnitVersion/TestData/HearingTestData.cs:51]: Please add a description to the method so it is clear why the returning object differs to the method above.
Also the name might be more meaningful as `CreateHearingWithAnEndpoint`.

File: UI.NUnitVersion/TestData/HearingTestData.cs [hunk 2]
```diff
@@ -47,4 +47,43 @@ public static BookingDto CreateHearingDto(bool remote = false, DateTime? schedul
...
+            DurationMinute = 30,
+            VenueName = "Birmingham Civil and Family Justice Centre",
+            RoomName = "Room 1",
+            Judge = new BookingJudgeDto
+            {
+                Username = "auto_aw.judge_02@hearings.reform.hmcts.net",
+                DisplayName = "Auto Judge",
+                Phone = ""
+            },
+            Participants = new List<BookingExistingParticipantDto>
```
Comment (shaed-parkar) [inline UI.NUnitVersion/TestData/HearingTestData.cs:72]: Where are the list of endpoints?

File: UI.NUnitVersion/Admin/Booking/BookingHearingWithEndPointTest.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        hearingSchedulePage.EnterSingleDayHearingSchedule(bookingDto.ScheduledDateTime, bookingDto.DurationHour,
+            bookingDto.DurationMinute, bookingDto.VenueName, bookingDto.RoomName);
+        
+        var assignJudgePage = hearingSchedulePage.GoToNextPage();
+        assignJudgePage.EnterJudgeDetails("auto_aw.judge_02@hearings.reform.hmcts.net", "Auto Judge", "");
+
+        var addParticipantPage = assignJudgePage.GoToParticipantsPage();
+        addParticipantPage.AddExistingParticipants(bookingDto.Participants);
+        addParticipantPage.GoToVideoAccessPointsPage();
```
Comment (shaed-parkar) [inline UI.NUnitVersion/Admin/Booking/BookingHearingWithEndPointTest.cs:34]: The video endpoints should be added in the video endpoints page. I think you've misunderstood the concept of page object models, you've not used the returning object here to navigate through the pages

File: UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs [hunk 4]
```diff
@@ -154,4 +158,17 @@ public VideoAccessPointsPage GoToVideoAccessPointsPage()
...
         return new VideoAccessPointsPage(Driver, DefaultWaitTime);
     }
+    
+    public SummaryPage GoToSummaryPage()
+    {
+        ClickElement(_nextButton);
+        return new SummaryPage(Driver, DefaultWaitTime);
+    }
+    
+    public void CompleteBookAVideoHearing(string displayName, string advocate)
```
Comment (shaed-parkar) [inline UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:168]: This has nothing to do with completing a booking. This is the Participant's page, where only managing the participant list should be concerned

File: UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs [hunk 5]
```diff
@@ -154,4 +158,17 @@ public VideoAccessPointsPage GoToVideoAccessPointsPage()
         ClickElement(_nextButton);
         return new VideoAccessPointsPage(Driver, DefaultWaitTime);
     }
+    
+    public SummaryPage GoToSummaryPage()
```
Comment (shaed-parkar) [inline UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:162]: When would a use want to go to the SummaryPage from the participant page in the journey in your test?

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
Let's have a chat about page object models to avoid the same style as the spec flow project else there is no point of re-writing the tests

File: UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs (hunk 5 above)
Comment (phildebrah) [reply UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:162]: there is a summary page to drive participant to next

Comment (phildebrah) [reply UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:162]: var summaryPage = addParticipantPage.GoToSummaryPage();

File: UI.NUnitVersion/TestData/HearingTestData.cs (hunk 1 above)
Comment (phildebrah) [reply UI.NUnitVersion/TestData/HearingTestData.cs:51]: Thus the Participant date for end point

File: UI.NUnitVersion/TestData/HearingTestData.cs (hunk 2 above)
Comment (phildebrah) [reply UI.NUnitVersion/TestData/HearingTestData.cs:72]: BookingExistingParticipantDto.Representative(GenericTestParty.Defendant, GenericTestRole.Representative,
                    "auto_vw.representative_157@hmcts.net", "auto_vw.representative_157@hearings.reform.hmcts.net",
                    "EndPoint 4",
                    "EndPoint 3")

File: UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs (hunk 4 above)
Comment (phildebrah) [reply UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:168]: we can move it from this Participant page class but you have place Video access element in that section

File: UI.NUnitVersion/Admin/Booking/BookingHearingWithEndPointTest.cs (hunk 3 above)
Comment (phildebrah) [reply UI.NUnitVersion/Admin/Booking/BookingHearingWithEndPointTest.cs:34]: there is a method  to navigate to Video point- Participant page Class
public void CompleteBookAVideoHearing(string displayName, string advocate)
    {
        EnterText(DisplayName(0), displayName);
        SelectDropDownByText(DefenceAdvocate(0), advocate);
        ClickElement(_nextButton);
    }

File: UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs (hunk 5 above)
Comment (phildebrah) [reply UI.PageModels/Pages/Admin/Booking/ParticipantsPage.cs:162]: please let have chat when ever you are free

**Pull request closed without merging**